check_command.go

```diff
@@ -34,12 +34,29 @@ func (c *CheckCommand) Run(request CheckRequest) ([]Version, error) {
 	var latestVersions []Version
 
 	for _, deployment := range deployments {
+		if len(request.Source.Environments) > 0 {
+			found := false
+			for _, env := range request.Source.Environments {
+				if env == *deployment.Environment {
+					found = true
+				}
+			}
+
+			if !found {
+				continue
+			}
+		}
+
 		id := *deployment.ID
 		if strconv.Itoa(id) >= request.Version.ID {
 			latestVersions = append(latestVersions, Version{ID: strconv.Itoa(id)})
 		}
 	}
 
+	if len(latestVersions) == 0 {
```

Comment:
Shall we remove the check for `length == 0` on lines 30-32 now, in favour of this single check here?